internal/core/app/app.go

```diff
@@ -0,0 +1,92 @@
+package app
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/arturzhamaliyev/Flight-Bookings-API/internal/core/logging"
+	"go.uber.org/zap"
+)
+
+// Listener represents a type that can listen for incoming connections.
+type Listener interface {
+	Listen(ctx context.Context) error
+}

```

Comment:
why do you created interface for listener?

Reply:
i only needed Listen() method from custom http package to start an app